### ISSUE TITLE
Better looking connected glass

### ISSUE THREAD

**Issue by AFCMS** [minetest/minetest#11546]
## Problem

There is three problems:

* Glass panes cant use this draw type.
* The 3d border of glass blocks is ugly
* The inventory image of glass block is weird looking ans should be exactly like if connected glass is disabled.

In minetest:

![screenshot_20210817_084316](https://user-images.githubusercontent.com/61794590/129722880-96989f9e-7a54-4404-a564-118b2e9d83a3.png)

In minecraft with optifine:
![image](https://user-images.githubusercontent.com/61794590/129730650-80acda4c-dd64-4909-8e24-67f20ba90acb.png)

![image](https://user-images.githubusercontent.com/61794590/129723327-528af891-ec79-4f59-a19c-09c0fbec9679.png)
The optifine glass is way better.

## Solutions

Allow glass panes to use this drawtype.
Remove glass 3d border.
Make inventory image the same for connected glass and normal glass.

**Comment by TuringZauber** [comment]
MC2 would also benefit from this: https://git.minetest.land/MineClone2/MineClone2/issues/1817

**Comment by numberZero** [comment]
> Allow glass panes to use this drawtype.

Not sure what do you mean.

> Remove glass 3d border, just make connected glass textures seamless.

Seamless textures aren’t that simple. Even if we only consider 4 direct neighbors (left, top, right, and bottom) that’s 16 (2^4) images while we have 12 at most (6 tiles + 6 special tiles). And even that will look buggy at inner corners:
![image](https://user-images.githubusercontent.com/7352626/228966159-d2831fcb-e1bb-4221-a3a2-a843ab363040.png)
Some packing is possible though, e.g. using a tile per frame side, and render several one over another as necessary; so the textures would look like:
![image](https://user-images.githubusercontent.com/7352626/228967856-2cbc59ab-b6a6-4443-ac0a-45aa14d8b9e5.png)
It is possible to add another 4 tiles for inner corners but that would require using the regular tiles in an unusual way.

**Comment by sfan5** [comment]
> The 3d border of glass blocks is ugly

This is entirely subjective, I think it looks good.

**Comment by fluxionary** [comment]
46, lovely. the combinatorics here are interesting, but not  directly relevant to minetest.

**Comment by numberZero** [comment]
> > The 3d border of glass blocks is ugly
> 
> This is entirely subjective, I think it looks good.

Yes it is subjective.

> 46 separate textures

47. But trying to stuff that in 12 tiles is something I definitely don’t want to do; not until GL3 and its array textures at least.

> but for stained glass, that wouldn't work, since the stained glass would become very opaque from all the overlaying.

Assuming stained glass with opaque frame, that *would* work: overlays would need to be completely transparent except of the frame.

Now, implementing it this way is possible, and reasonably easy. But only a core dev can answer the critical question:
**Does it worth a new drawtype?** (and review and future maintenance effort)

If yes, I implement this.
If no, close the issue as “wontfix”.

**Comment by Desour** [comment]
>The inventory image of glass block is weird looking ans should be exactly like if connected glass is disabled.

Node inventory items look like the placed node, unless specified otherwise by the mod. Why should glasslike be an exception here?
(This may also be subjective.)

-----------

Connected textures is essentially a duplicate of #8334.

----

>Glass panes cant use this draw type.

xpanes are currently implemented with connected nodeboxes (see <https://github.com/minetest/minetest_game/blob/master/mods/xpanes/init.lua>).

A good way to implement this would be to add something much more flexible, for example:
Multiple connection groups (for glass plane to other other glass plane vs. glass plane to solid node). And extend the nodebox format to include something like a tileset index for each box. Or connected meshes (do we have an issue for this?).

**Comment by numberZero** [comment]
> Connected textures is essentially a duplicate of #8334.

Or rather, of #2748.

> A good way to implement this would be to add something much more flexible, for example:
> Multiple connection groups (for glass plane to other other glass plane vs. glass plane to solid node). And extend the nodebox format to include something like a tileset index for each box. Or connected meshes (do we have an issue for this?).

Hell no. Increasing complexity is not a solution. And you *would* need the complexity at least to stuff 47 images in 12 tiles in a generic way.

**Comment by Desour** [comment]
> > A good way to implement this would be to add something much more flexible, for example:
> > Multiple connection groups (for glass plane to other other glass plane vs. glass plane to solid node). And extend the nodebox format to include something like a tileset index for each box. Or connected meshes (do we have an issue for this?).
> 
> Hell no. Increasing complexity is not a solution.

Well, I guess a new specialized pane drawtype would also be justified. `¯\_(ツ)_/¯`

> And you _would_ need the complexity at least to stuff 47 images in 12 tiles in a generic way.

What would be holding us back from just using more triangles? That would probably just need two different tiles or so.

**Comment by numberZero** [comment]
Sorry for being harsh.

> a new specialized pane drawtype

Oops, missed that. My proposal was for full nodes only. Things can get hairy this way.

Anyway, my argument is: if this can’t or shouldn’t be implemented easily, it doesn’t worth the effort of even discussing it.
